MoltenVK/CMakeLists.txt

```diff
@@ -0,0 +1,204 @@
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Commands/MVKCommandResourceFactory.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Commands/MVKMTLBufferAllocation.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKBuffer.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKDescriptor.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKDescriptorSet.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKDevice.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKDeviceMemory.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKFramebuffer.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKImage.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKInstance.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKPipeline.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKPixelFormats.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKQueryPool.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKQueue.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKRenderPass.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKResource.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKShaderModule.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKSurface.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKSwapchain.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKSync.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/GPUObjects/MVKVulkanAPIObject.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Layers/MVKExtensions.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Layers/MVKLayers.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/CAMetalLayer+MoltenVK.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/MTLRenderPassDepthAttachmentDescriptor+MoltenVK.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/MTLRenderPassDescriptor+MoltenVK.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/MTLRenderPassStencilAttachmentDescriptor+MoltenVK.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/MTLRenderPipelineDescriptor+MoltenVK.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/MTLSamplerDescriptor+MoltenVK.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/MTLTextureDescriptor+MoltenVK.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/MVKGPUCapture.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/OS/NSString+MoltenVK.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Utility/MVKBaseObject.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Utility/MVKBlockObserver.m
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Utility/MVKCodec.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Utility/MVKEnvironment.cpp
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Utility/MVKFoundation.cpp
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Utility/MVKWatermark.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Vulkan/mvk_datatypes.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Vulkan/mvk_api.mm
+        ${CMAKE_CURRENT_LIST_DIR}/MoltenVK/Vulkan/vulkan.mm
+        )
```

Comment:
I'm definitely no `CMake` expert, but is it possible to generate the above file lists automatically from the directories? 

My concern is that these lists will get out of date if contributors add a file to MoltenVK, and forget to update these lists.

And I wonder if we could add running a `CMake` build to `CI.yml`, so it gets tested during GitHub CI for future PR submissions, and catch issues such as these lists being out of sync?

Reply:
Sure, we can use GLOB instead.  The advantage to GLOB is that it detects all files dynamically as you'd expect here.  The disadvantage is it can be easy to get what you're asking for and not what you mean ;-).  In common practice CMake projects list out all files in the manner here.  However, switching to GLOB is not a problem ultimately.

Reply:
I hear you on the "be careful what you ask for". 

Currently the focus for internal development is Xcode. As long as that remains primary, the `CMake` will have to follow it, and focusing on directories might make the best option for consistency. If we move to your larger recommendation, where `CMake` is primary, and the Xcode editing environment is generated from it, them listing the files out will make the most sense.

Reply:
This is now switched to GLOB to provide exactly as asked for results.